multiversx/mx-chain-go
# Mx 13965 block processing cutoff

## Reasoning behind the pull request
- there are use cases when one would like a node to stop advancing or reprocess the same block at a given coordinate (round nonce or epoch)
  
## Proposed changes
- added a new feature called `block processing cutoff` that can be configured to pause or reprocess at a specific nonce/round/epoch

## Testing procedure
Inside `prefs.toml`'s `BlockProcessingCutoff` section, set `Enabled` to `true` and update the rest of the config with variations of cutoff types and values.
1. `process-error` mode: the node should re-process the same block over and over
2. `pause`: the node should not advance

The testing can be done via calling the Rest API endpoints (for example `/node/status` and check for nonce to remain the same in case of pause, and in case or process-error it could roll-back 1 block).
Also, by analyzing the logs, the same behaviour should be observed

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?
